crates/rethnet/src/config.rs

```diff
@@ -31,31 +32,124 @@ pub const DEFAULT_PRIVATE_KEYS: [&str; 20] = [
 ];
 
 /// struct representing the deserialized conifguration file, eg hardhat.config.json
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
 pub struct ConfigFile {
     // TODO: expand this per https://github.com/NomicFoundation/rethnet/issues/111
-    pub accounts: Vec<AccountConfig>,
-    pub chain_id: u64,
-    pub coinbase: Address,
-    pub network_id: u64,
+    pub accounts: Option<Vec<AccountConfig>>,
```

Comment:
I _think_ not, because isn't it the case that the serde defaults are only applied when you're actually deserializing from something? Because we support the case of having _no_ config files (and, note, in that case we're depending on `ConfigFile::default()` to produce the default values for us) I don't think serde defaults would work. Does that make sense? Or am i missing something?

Reply:
You would need to remove all options:

```rs
pub accounts: Vec<AccountConfig>,
#[serde(default)]
pub block_gas_limit: U256,
...
```

and then implement `trait Default` with all values set to the default values, without `Some`

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/4221/commits/1e8b0423084a01596683109ba7b65fe44e79ba49

Reply:
There is no need for separate functions, you can just use the the `Default` trait implementation: https://serde.rs/field-attrs.html#default

Reply:
Thanks for pushing this further. I had thought that `serde` would need each field to have its own separate defaults specified, and the only way to do that at the field level (dynamically anyways) is with such functions. But I wrote a little test to prove to myself that serde can handle some user specified values mixed in with defaults, which is included in the commit that addresses this, https://github.com/NomicFoundation/hardhat/pull/4221/commits/813abdea7381610ee9746aa2bbb338c83dbb428d